rotkehlchen/globaldb/handler.py

```diff
@@ -298,35 +298,53 @@ def get_assets_mappings(identifiers: List[str]) -> Dict[str, dict]:
         return result
 
     @staticmethod
-    def search_assets(filter_query: 'AssetsFilterQuery') -> List[Dict[str, str]]:
+    def search_assets(filter_query: 'AssetsFilterQuery', db: 'DBHandler') -> List[Dict[str, str]]:
         """Returns a list of asset details that match the search query provided."""
         search_result = []
         query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)  # noqa:E501
-        with GlobalDBHandler().conn.read_ctx() as cursor:
-            cursor.execute(query, bindings)
-            for entry in cursor:
-                search_result.append({
-                    'identifier': entry[0],
-                    'name': entry[1],
-                    'symbol': entry[2],
-                })
+        with db.conn.read_ctx() as cursor:
+            with GlobalDBHandler().conn.read_ctx() as global_db_cursor:
```

Comment:
Tell me if I'm wrong but I believe we only use this in the rest.py file right? You could pass here the cursor and make this logic less indented. Also you can call the `with` with the two cursors in the same line. I believe our style is passing the db or if we can reuse then pass the cursor

Reply:
<img width="892" alt="image" src="https://user-images.githubusercontent.com/72208758/192266322-29ee1568-cef7-4953-b329-5a73d844d3e9.png">

Only one usage of `with GlobalDBHandler().conn.read_ctx() as cursor:` is present in `rest.py`. The usage was added by my previous pr.

Reply:
Yeah, I'll move the context managers to the same line